metaflow/plugins/events_decorator.py

```diff
@@ -96,47 +129,69 @@ def process_event_name(self, event):
                 "'parameters': {'alpha': 'beta'}})"
             )
 
-    def process_parameters(self, parameters):
+    def process_parameters(self, parameters, event_name):
+        """
+        Process the parameters for an event and return a dictionary of parameter mappings if
+        parameters was statically defined or a function if deploy-time trigger.
+
+        Parameters
+        ----------
+        Parameters : Union[callable, List[Union[str, Tuple[str, str]]], Dict[str, str]]
+            Parameters to process
+
+        event_name : Union[str, callable]
+            Name of the event
+
+        Returns
+        -------
+        Union[Dict[Union[str, callable], Union[str, callable]], Callable]
```

Comment:
return type seems too permissive. You are either returning a callable or a Dict[str, str]